File: Cpp/Unique Paths.cpp
class Solution {
public:
    /**
     * @param n, m: positive integer (1 <= n ,m <= 100)
     * @return an integer
     */
    int uniquePaths(int m, int n) {
        // wirte your code here
        vector<vector <int> > ivec(m ,vector<int>(n));
        ivec[0][0] = 1;
        for (int i = 0; i < m; i ++) {
            for (int j = 0; j < n; j ++) {
                if (i != 0 || j != 0) {
                    if (i == 0) ivec[i][j] = ivec[i][j-1];
                    else if (j == 0) ivec[i][j] = ivec[i-1][j];
                    else ivec[i][j] = ivec[i-1][j] + ivec[i][j-1];
                }
            }
        }
        return ivec[m-1][n-1];
    }
};

File: Java/Unique-Paths.java
public class Solution {
    /**
     * @param n, m: positive integer (1 <= n ,m <= 100)
     * @return an integer
     */
    public int uniquePaths(int m, int n) {
        // write your code here
        int dp[ ][ ] = new int[m][n];
         for (int i = 0; i < n; i ++) {
         	dp[0][i] = 1;
         }
         for (int i = 0; i < m; i ++) {
         	dp[i][0] = 1;
         }
         for (int i = 1; i < m; i ++) {
         	for (int j = 1; j < n; j ++) {
         		dp[i][j] = dp[i - 1][j] + dp[i][j - 1];
         	}
         }
         return dp[m - 1][n - 1];
    }
}

